node/nodeTransactions.go

```diff
@@ -0,0 +1,194 @@
+	if found {
+		return n.unmarshalTransaction(txBytes, txType)
+	}
+
+	txBytes, txType, found = n.getTxBytesFromStorage(hash)
+	if found {
+		return n.unmarshalTransaction(txBytes, txType)
+	}
+
+	return nil, fmt.Errorf("transaction not found")
+}
+
+// GetTransactionStatus gets the transaction status
+func (n *Node) GetTransactionStatus(txHash string) (string, error) {
+	if !n.apiTransactionByHashThrottler.CanProcess() {
+		return "", ErrSystemBusyTxHash
+	}
+
+	n.apiTransactionByHashThrottler.StartProcessing()
+	defer n.apiTransactionByHashThrottler.EndProcessing()
+
+	hash, err := hex.DecodeString(txHash)
+	if err != nil {
+		return "", err
+	}
+
+	_, _, foundInDataPool := n.getTxBytesFromDataPool(hash)
+	if foundInDataPool {
+		return string(core.TxStatusReceived), nil
+	}
+
+	foundInStorage := n.isTxInStorage(hash)
+	if foundInStorage {
+		return string(core.TxStatusExecuted), nil
+	}
+
+	return string(core.TxStatusUnknown), nil
+}
+
+func (n *Node) getTxBytesFromDataPool(hash []byte) ([]byte, transactionType, bool) {
+	txsPool := n.dataPool.Transactions()
+	txBytes, found := txsPool.SearchFirstData(hash)
+	if found && txBytes != nil {
+		return txBytes.([]byte), normalTx, true
+	}
+
+	rewardTxsPool := n.dataPool.RewardTransactions()
+	txBytes, found = rewardTxsPool.SearchFirstData(hash)
+	if found && txBytes != nil {
+		return txBytes.([]byte), rewardTx, true
+	}
+
+	unsignedTxsPool := n.dataPool.UnsignedTransactions()
+	txBytes, found = unsignedTxsPool.SearchFirstData(hash)
+	if found && txBytes != nil {
+		return txBytes.([]byte), unsignedTx, true
+	}
+
+	return nil, invalidTx, false
+}
+
+func (n *Node) isTxInStorage(hash []byte) bool {
+	txsStorer := n.store.GetStorer(dataRetriever.TransactionUnit)
+	err := txsStorer.Has(hash)
+	if err == nil {
+		return true
+	}
+
+	rewardTxsStorer := n.store.GetStorer(dataRetriever.RewardTransactionUnit)
+	err = rewardTxsStorer.Has(hash)
+	if err == nil {
+		return true
+	}
+
+	unsignedTxsStorer := n.store.GetStorer(dataRetriever.UnsignedTransactionUnit)
+	err = unsignedTxsStorer.Has(hash)
+	return err == nil
+}
+
+func (n *Node) getTxBytesFromStorage(hash []byte) ([]byte, transactionType, bool) {
+	txsStorer := n.store.GetStorer(dataRetriever.TransactionUnit)
+	txBytes, err := txsStorer.SearchFirst(hash)
+	if err == nil {
+		return txBytes, normalTx, true
+	}
+
+	rewardTxsStorer := n.store.GetStorer(dataRetriever.RewardTransactionUnit)
+	txBytes, err = rewardTxsStorer.SearchFirst(hash)
+	if err == nil {
+		return txBytes, rewardTx, true
+	}
+
+	unsignedTxsStorer := n.store.GetStorer(dataRetriever.UnsignedTransactionUnit)
+	txBytes, err = unsignedTxsStorer.SearchFirst(hash)
+	if err == nil {
+		return txBytes, unsignedTx, true
+	}
+
+	return nil, invalidTx, false
+}
+
+func (n *Node) unmarshalTransaction(txBytes []byte, txType transactionType) (*transaction.ApiTransactionResult, error) {
+	switch txType {
+	case normalTx:
+		var tx transaction.Transaction
```

Comment:
make you can make a method for every type of transaction ...unmarshalNormalTx .... etc

Reply:
no need for that.. it is just unmarshaling + creating an object